zetaclient/chains/solana/signer/signer.go

```diff
@@ -213,6 +215,27 @@ func (signer *Signer) TryProcessOutbound(
 	// set relayer balance metrics
 	signer.SetRelayerBalanceMetrics(ctx)
 
+	isNonceRelayable := signer.waitNonceRelayable(ctx, nonce, logger)
```

Comment:
If you really want a matching Solana `pending nonce`, you can do it right [here](https://github.com/zeta-chain/node/blob/solana-pda-nonce-preflight-v2/zetaclient/chains/solana/signer/signer.go#L323) very simply in the tx broadcaster logic, and we're already waiting for up to `64` seconds in there.

Reply:
Note that I'm using confirmation:processed not the default confirmation:finalized which that code is using. I think I'd actually need to update this code a bit to not give up when `nonce > pda.Nonce` when `confirmation:processed` as the nonce could go backwards on a failure

Reply:
You can just switch the commitment level to `CommitmentProcessed` to give it a try. We must face a harsh reality: there is no accurate way to tell the exact pending nonce in an off-chain program (here is `zetaclientd`).

- zetaclient may connect to a load balancer, which wraps a group of endpoints.
- each RPC endpoint have their own Solana chain state which may not be in good sync.

The only thing we can do is to retry and optimize the timing, leaving the Solana endpoint itself resolve all other conflicts.

Reply:
I both of those are very valid concerns but I think we could also try to solve those at the infrastructure level by selecting a higher quality RPC provider.

Reply:
Do note that this pda nonce check loop happens before the relayer transaction signing so that we can avoid starting the two minute recent blockhash timer until we're in a valid range.